### DISCUSSION TITLE
gatsby-plugin-image does not take baseUrl option from jsconfig.json

### DISCUSSION THREAD

**Discussion by lnikell** [gatsbyjs/gatsby#30700] [Ideas / Feature Requests]
## Description

The Gatsby default configuration works well with jsconfig.json and baseUrl option such as this 

```
{
  "compilerOptions": {
    "baseUrl": "src",
    "jsx": "react"
  }
}

```
So you don't need to write always relative paths and can use `components/my-example` paths. 

We tried to migrate to Gatsby V3 and wanted to use paths to previously default root `images`(as of v2), but we don't wan't to write relative paths to those images such as "../../../images/example.png" and want to use advantages of jsconfig.json and baseUrl and have an option to write just "images/example.png".

### Expected behavior

gatsby-plugin-image should be aware of baseUrl configuration and load images accordingly.

**Comment by ascorbic** [comment]
We're unlikely to add this, because the component doesn't use module resolution at all. The idea is to be as simple as possible, so it uses filesystem paths and nothing else.

**Reply by lnikell** [reply]
Thank you @ascorbic! For the time being for those who suffer from this, I found the workaround to use GatsbyImage instead of StaticImage, so when you migrate to a new version you just have to change the query a little bit and use getImage. In that case you can easily continue to fetch images from root images folder as of gatsby v2
